controllers/networkfence_controller.go

```diff
@@ -113,37 +114,78 @@ func (r *NetworkFenceReconciler) Reconcile(ctx context.Context, req ctrl.Request
 		return ctrl.Result{}, err
 	}
 
-	// check if the networkfence object is getting deleted, if yes, then send
-	// UnfenceCluster request to the driver.
-	// After the UnfenceCluster request is processed, the finalizer will be removed
-	// so that the networkfence object can be deleted gracefully.
-	if !nwFence.GetDeletionTimestamp().IsZero() {
+	nf := NetworkFenceInstance{
+		reconciler:       r,
+		logger:           logger,
+		instance:         nwFence,
+		controllerClient: client,
+	}
+
+	// check if the networkfence object is getting deleted and handle it.
+	if !nf.instance.GetDeletionTimestamp().IsZero() {
 		if util.ContainsInSlice(nwFence.GetFinalizers(), networkFenceFinalizer) {
 
-			err := r.unfenceClusterNetwork(ctx, nwFence, client, logger)
+			err := nf.removeFinalizerFromNetworkFence(ctx)
 			if err != nil {
-				logger.Error(err, "failed to unfence cluster network")
+				logger.Error(err, "failed to remove finalizer on NetworkFence resource")
 				return ctrl.Result{}, err
 			}
 		}
+
 		logger.Info("NetworkFence object is terminated, skipping reconciliation")
 		return ctrl.Result{}, nil
 	}
 
-	err = r.fenceClusterNetwork(ctx, nwFence, client, logger)
+	if nwFence.Spec.FenceState == csiaddonsv1alpha1.Fenced {
+		nf.logger.Info("FenceClusterNetwork Request")
+	} else {
+		nf.logger.Info("UnFenceClusterNetwork Request")
+	}
```

Comment:
log namespaced name (CR name and namespace name) which helps

Reply:
ACK